### PR TITLE
cli: Add minimal `rad comment`

### PR DESCRIPTION
**Pull request by slack-coder**
Allow users to comment on an Issue.

References: #187

### CODE REVIEW

File: radicle-cli/src/commands/comment.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    if message.is_empty() {
+        return Ok(());
+    }
+
+    let mut issues = Issues::open(*signer.public_key(), repo)?;
+    let mut issue = issues.get_mut(&options.id).map_err(|e| match e {
+        store::Error::NotFound(_, _) => anyhow::anyhow!("Could not find issue {}", options.id),
+        _ => e.into(),
+    })?;
+    let comment_id = *issue.root().expect("root comment always exists").0;
```
Comment (cloudhead) [inline radicle-cli/src/commands/comment.rs:101]: We usually prefer deconstructing tuples than using `.0` etc., I'd do `let (comment_id, _) = ...`

File: radicle-cli/src/commands/comment.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        ))
+    }
+}
+
+fn comment(
+    options: &Options,
+    repo: &storage::git::Repository,
+    signer: impl Signer,
+) -> anyhow::Result<()> {
+    let message = options.message.clone().get("Enter a comment message...");
```
Comment (cloudhead) [inline radicle-cli/src/commands/comment.rs:91]:
```suggestion
    let message = options.message.clone().get("Enter a comment...");
```

File: radicle-cli/src/commands/comment.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                Long("help") => return Err(Error::Help.into()),
+
+                Value(val) if id.is_none() => id = Some(parse_cob_id(val)?),
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("a patch id to merge must be provided"))?,
```
Comment (cloudhead) [inline radicle-cli/src/commands/comment.rs:78]: error text needs updating

File: radicle-cli/src/commands/comment.rs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad comment <id> [-m <text>] [--reply-to <index>]
+
+Options
+
+    -m, --message               Comment message
+        --reply-to <index>      Index of comment writing a reply for
```
Comment (cloudhead) [inline radicle-cli/src/commands/comment.rs:29]: `reply-to` doesn't seem supported yet.

**Comment by cloudhead** [comment]
Looks good overall. Would be good to have a test at some point that checks that the comment was recorded properly, and also tests passing multiple `-m` arguments. This could be a separate pr.

**Comment by slack-coder** [comment]
Thanks :) all points addressed.  Indeed a few follow up PRs are needed.

File: radicle-cli/src/commands/comment.rs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                Long("help") => return Err(Error::Help.into()),
+
+                Value(val) if id.is_none() => id = Some(parse_cob_id(val)?),
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("an issue id to merge must be provided"))?,
```
Comment (cloudhead) [inline radicle-cli/src/commands/comment.rs:77]:
```suggestion
                id: id.ok_or_else(|| anyhow!("an issue id to comment on must be provided"))?,
```

**Pull request merged**